sdk/eventhub/event-hubs/test/receiver.spec.ts

```diff
@@ -891,123 +874,48 @@ describe("EventHub Receiver", function(): void {
   });
 
   describe("Negative scenarios", function(): void {
-    describe("on invalid partition ids like", function(): void {
-      const invalidIds = ["XYZ", "-1", "1000", "-"];
-      invalidIds.forEach(function(id: string): void {
-        it(`"${id}" should throw an error`, async function(): Promise<void> {
-          try {
-            debug("Created receiver and will be receiving messages from partition id ...", id);
-            const d = await client
-              .createConsumer(EventHubConsumerClient.defaultConsumerGroupName, id, latestEventPosition)
-              .receiveBatch(10, 3);
-            debug("received messages ", d.length);
-            throw new Error("Test failure");
-          } catch (err) {
-            debug("Receiver received an error", err);
-            should.exist(err);
-            err.message.should.match(
-              /.*The specified partition is invalid for an EventHub partition sender or receiver.*/gi
-            );
-          }
-        });
+    it("should throw MessagingEntityNotFoundError for non existing consumer group", async function(): Promise<
+      void
+    > {
+      const badConsumerClient = new EventHubConsumerClient(
+        "boo",
+        service.connectionString,
+        service.path
+      );
+      let subscription: Subscription | undefined;
+      const caughtErr = await new Promise<Error | MessagingError>((resolve) => {
+        subscription = badConsumerClient.subscribe(
+          {
+            processEvents: async () => {},
+            processError: async (err) => {
+              resolve(err);
+            }
+          },
+          { maxWaitTimeInSeconds: 2 }
```

Comment:
What is special about 2 seconds? Is it really that precise?

Given the history of how tests bounce in CI depending on perf I'm nervous to see hardcoded timeouts (and especially short ones).

I don't recall the default for receiveBatch but I'm guessing it was 60 seconds?

Reply:
This is because the program would exit only after the 1 min timer runs out. Should be fixed after https://github.com/Azure/azure-sdk-for-js/pull/9543.

For now, I updated it to 0 and added a TODO comment to remove it once https://github.com/Azure/azure-sdk-for-js/pull/9543 is merged

Reply:
Awesome!